jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -970,38 +971,49 @@ public boolean checkVirtualHost(final Request baseRequest)
         if (_vhosts != null && _vhosts.length > 0)
         {
             String vhost = normalizeHostname(baseRequest.getServerName());
-
-            boolean match = false;
-            boolean connectorName = false;
-            boolean connectorMatch = false;
-
-            for (String contextVhost:_vhosts)
+            String connectorName = baseRequest.getHttpChannel().getConnector().getName();
+            for (String contextVhost : _vhosts)
             {
-                if (contextVhost == null || contextVhost.length()==0)
+                if (contextVhost == null || contextVhost.length() == 0)
                     continue;
-                char c=contextVhost.charAt(0);
-                switch (c)
+
+                int connectorPos = contextVhost.indexOf('@') + 1;
+                if (connectorPos > 0)
                 {
-                    case '*':
-                        if (contextVhost.startsWith("*."))
-                            // wildcard only at the beginning, and only for one additional subdomain level
-                            match = match || contextVhost.regionMatches(true,2,vhost,vhost.indexOf(".") + 1,contextVhost.length() - 2);
-                        break;
-                    case '@':
-                        connectorName=true;
-                        String name=baseRequest.getHttpChannel().getConnector().getName();
-                        boolean m=name!=null && contextVhost.length()==name.length()+1 && contextVhost.endsWith(name);
-                        match = match || m;
-                        connectorMatch = connectorMatch || m;
-                        break;
-                    default:
-                        match = match || contextVhost.equalsIgnoreCase(vhost);
+                    // If there is no connector name and we have a match for connector only
+                    // we fail to match.  If we have connector and host we can continue
+                    // to check for other host only matches.
+                    if (connectorName == null || connectorName.length() == 0 )
+                    {
+                        if (connectorPos > 1)
+                            continue;
+                        else
+                            return false;
+                    }
+                    // If connector specifier in vhost entry and no match then continue to next entry
+                    // no need to check host
+                    if (!contextVhost.regionMatches(true,connectorPos,connectorName,0,connectorName.length()))
+                        continue;
+                    // If we matched connector and no host portion was specified we match
+                    if (connectorPos == 1)
+                        return true;
                 }
-
+                // At this point either connector was specified and there was a connector match,
```

Comment:
`At this point either no connector was specified or there was a connector match...`

Reply:
I will fix this